tests/compile-fail/panic/bad_unwind.rs

```diff
@@ -4,6 +4,8 @@
 //! Unwinding when the caller ABI is "C" (without "-unwind") is UB.
 //! Currently we detect the ABI mismatch; we could probably allow such calls in principle one day
 //! but then we have to detect the unexpected unwinding.
+//! FIXME: `-Zmiri-disable-abi-check` does not work for this test because function pointers are
+//! always allowed to unwind.
```

Comment:
This comment doesn't make sense to me... the flag still works, right?

Reply:
I actually mean the flag doesn't *manage to* resolve the issue in this test. The flag itself still works, and it makes this test `run-pass`. Sorry for the sloppy wording here.

Reply:
Where "manage to resolve" here is "if the abort-on-panic shim is added this test will not work any more"?

Well, the abort-on-panic shim successfully does away with the UB this is testing for, so that seems fine. ;) We can just remove the test then. We still have your other tests based on the allocator-nounwind attribute.

Reply:
> We can just remove the test then. We still have your other tests based on the allocator-nounwind attribute.

Fair. I removed the comment for now.